### PR TITLE
Completed field label conversion funcs for Namespace

### LINKED ISSUE #5462: Missing field selector conversions
https://github.com/kubernetes/kubernetes/pull/15766 added field selectors for namespace to several objects, but did not add conversions for those field selectors to api/v1/conversion.go#AddFieldLabelConversionFunc.

I'm pretty sure that means the new selectors aren't usable. Please add those conversions, and add tests for using the selectors.

### PR DESCRIPTION
**Pull request by miminar**
- New selectable fields of Namespace added upstream:
  - "metadata.name"
  - "name"

which were not covered by field label conversion functions.

Resolves #5462

### CODE REVIEW

**Comment by miminar** [comment]
@liggitt can you please take a look before I resubmit it upstream?

File: Godeps/_workspace/src/k8s.io/kubernetes/pkg/api/conversion_test.go [hunk 1]
```diff
@@ -107,3 +110,59 @@ func BenchmarkReplicationControllerConversion(b *testing.B) {
...
+		{"status", "", true},
+	}
+
+	labels := namespace.NamespaceToSelectableFields(ns)
+	lSet := sets.NewString()
+	for l := range labels {
+		lSet.Insert(l)
+	}
+
+	for _, apiVersion := range []string{"v1", "v1beta3"} {
```
Comment (liggitt) [inline Godeps/_workspace/src/k8s.io/kubernetes/pkg/api/conversion_test.go]: don't hard-code versions, `use testapi.Default.Version()`

File: Godeps/_workspace/src/k8s.io/kubernetes/pkg/api/v1beta3/conversion.go [hunk 2]
```diff
@@ -126,7 +126,7 @@ func addConversionFuncs() {
 	err = api.Scheme.AddFieldLabelConversionFunc("v1beta3", "Namespace",
 		func(label, value string) (string, string, error) {
 			switch label {
-			case "status.phase":
+			case "name", "metadata.name", "status.phase":
```
Comment (liggitt) [inline Godeps/_workspace/src/k8s.io/kubernetes/pkg/api/v1beta3/conversion.go]: v1beta3 API is actually not enabled any more, so you can drop this

**Comment by liggitt** [comment]
drop the v1beta3 bits, fix the test, then submit upstream

**Comment by liggitt** [comment]
actually, didn't the upstream addition add metadata.namespace to lots of objects?

**Comment by miminar** [comment]
@liggitt  Yes. I've just added conversion functions for them too.

The upstream PR is [#16509](https://github.com/kubernetes/kubernetes/pull/16509).

File: Godeps/_workspace/src/k8s.io/kubernetes/pkg/api/v1beta3/conversion.go (hunk 2 above)
Comment (miminar) [inline Godeps/_workspace/src/k8s.io/kubernetes/pkg/api/v1beta3/conversion.go]: From this PR as well?

Comment (liggitt) [inline Godeps/_workspace/src/k8s.io/kubernetes/pkg/api/v1beta3/conversion.go]: yeah, field selectors are only used for querying/filtering... you can't query/filter v1beta3 any more

Comment (miminar) [inline Godeps/_workspace/src/k8s.io/kubernetes/pkg/api/v1beta3/conversion.go]: I see. Will do.

**Comment by miminar** [comment]
Removed `v1beta3` conversions additions and tests.

**Comment by liggitt** [comment]
squash/sync with upstream pull? not sure how urgently I feel like we need this

**Comment by miminar** [comment]
In that case we can close it here, can't we? I'm assuming the upstream PR will get merged.

**Comment by liggitt** [comment]
I would... I don't think we have an urgent call for it

**Pull request closed without merging**